### ISSUE TITLE
A problem about 315. Count of Smaller Numbers After Self

### ISSUE THREAD

**Issue by zyoung1024** [haoel/leetcode#229]
Hi, there.

In the Binary Index Tree implementation, the length of the tree should be one greater than the length of the original array (` nums `) . However,  your code is the same .

```c++
// @ 63-64 line
vector <int> f(sorted.size());
fenwick = f;
```

I think this will lead to some memory overflow issues. So, I modifies the size of `f` to `sorted.size() + 1`. It works in leetcode.

**Comment by Shanayshah44** [comment]
Yes Binary Index Tree is always one greater than the size of the array
